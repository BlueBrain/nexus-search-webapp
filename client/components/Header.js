import React from "react";
import logo from "../../public/img/logo.png";
import Login from "./Login";
import { Link } from "react-router-dom";
import SearchBar from "./SearchBar";

const HeaderComponent = () => (
  <div className="stack-wrap rounded-shadow stack-top">
  <header>
    <div className="logo-block">
      <Link
        className="logo bs-Button"
        to={"/home"}>
        <img src={logo} />
      </Link>
      <a
        className="fade in slow"
        href={"/"}>
        <h1>Search</h1>
      </a>
    </div>
    {/* <SearchBar /> */}
    <Login />
  </header>
  </div>
);

export default HeaderComponent;
