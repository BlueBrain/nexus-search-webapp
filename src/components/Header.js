import React from "react";
import logo from "../../public/img/logo.png";
import Login from "./Login";
import { WithStore } from "@bbp/nexus-react";
import { navigate } from "../store/actions";
import SearchBar from "./SearchBar";

const HeaderComponent = base => (
  <div className="stack-wrap rounded-shadow stack-top">
  <header>
    <div className="logo-block">
      <a className="logo bs-Button" href={base + "/home/"}>
        <img src={logo} />
      </a>
      <WithStore
        mapStateToProps={() => ({})}
        mapDispatchToProps={{
          navigate: navigate.navigate
        }}
      >
        {({ navigate }) => (
          <a
            href={base}
            onClick={e => {
              e.preventDefault();
              navigate("/");
            }}
          >
            {/* <h1>Search</h1> */}
          </a>
        )}
      </WithStore>
    </div>
    <SearchBar />
    <Login />
  </header>
  </div>
);

export default HeaderComponent;
