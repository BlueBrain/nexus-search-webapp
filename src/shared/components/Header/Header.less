@import '../../lib.less';

.Header {
  display: flex;
  justify-content: space-between;
  padding: 4px;
  box-sizing: content-box;
  height: 32px;
  background-color: @header-background-color;

  .logo-block {
    position: absolute;
    left: 50%;
    width: 120px;
    margin-left: -60px;

    .logo {
      margin-right: 0.5em;
      height: 32px;
      display: flex;
      align-items: center;

      img {
        height: 2em;
        width: 2em;
        margin: 1px 0;
        padding-bottom: 2px;
      }
    }

    h1 {
      color: @primary-color;
      margin: 0;
      font-weight: 100;
      display: inline-block;
    }

    @media (max-width: 795px) {
      left: 0;
      margin-left: 10px;
    }
  }

  .selectors {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    flex-grow: 1;
    margin-left: 10px;

    .OrgDropdown {
      height: 100%;
    }
  }

  .menu-block {
    display: flex;
    justify-content: flex-end;
    flex-grow: 1;
    padding-right: @default-pad;
    align-items: center;

    button,
    a {
      margin: 5px;
      font-weight: lighter;
    }

    .menu-dropdown {
      color: @text-color;
      align-self: center;
      margin-left: 10px;
    }
  }
}

.nav-item {
  font-size: 16px;
}

.popover-title {
  padding: 0 0 5px 0;
}

// the gtm modal should appear on a top of a popover
.ant-popover {
  z-index: 1000 !important;
}

/*
  ##Device = Tablets, Ipads (portrait)
  ##Screen = B/w 768px to 1024px
*/

@media (max-width: 1024px) {
  .hide-tablet {
    display: none;
  }
}
