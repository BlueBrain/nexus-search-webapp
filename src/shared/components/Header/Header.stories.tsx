import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withInfo } from '@storybook/addon-info';

import Header from './index';

storiesOf('Components/Header', module).add(
  'Header',
  withInfo(`
    The header component is used throughout the application.

    It is used to display the user name and navigation links

    ~~~js
      <Header
        name="Mark Hamill"
        links={[<a href="#">Log out</a>, <a href="#">Settings</a>]}
      />
      <Header onLoginClick={action('login-link-click')} />
      <Header displayLogin={false} />
    ~~~
  `)(() => (
    <React.Fragment>
      <div style={{ margin: '50px 40px 0px' }}>
        <h2>Name + menu + token</h2>
        <Header
          name="Mark Hamill"
          links={[<a href="#">Log out</a>, <a href="#">Settings</a>]}
          token="supercereal"
          githubIssueURL="some.url"
          version="v1.1.0"
        />
      </div>
      <div style={{ margin: '50px 40px 0px' }}>
        <h2>Name + menu</h2>
        <Header
          name="Mark Hamill"
          links={[<a href="#">Log out</a>, <a href="#">Settings</a>]}
          githubIssueURL="some.url"
          version="v1.1.0"
        />
      </div>
      <div style={{ margin: '50px 40px 0px' }}>
        <h2>No name + login link</h2>
        <Header
          onLoginClick={action('login-link-click')}
          githubIssueURL="some.url"
          version="v1.1.0"
        />
      </div>
      <div style={{ margin: '50px 40px 0px' }}>
        <h2>No name + no login link</h2>
        <Header
          displayLogin={false}
          githubIssueURL="some.url"
          version="v1.1.0"
        />
      </div>
    </React.Fragment>
  ))
);
