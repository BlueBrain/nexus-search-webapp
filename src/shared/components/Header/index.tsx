import * as React from 'react';
import { Menu, Dropdown, Icon, Button, Popover } from 'antd';
import { useSelector } from 'react-redux';

import Copy from '../Copy';
import ConsentPreferences from '../ConsentPreferences';
import { ConsentType } from '../../layouts/MainLayout';
import { RootState } from '../../store/reducers';

import './Header.less';

const logo = require('../../logo.svg');
const epflLogo = require('../../EPFL-logo.svg');

const documentationURL = 'https://bluebrainnexus.io/docs';

interface InformationContentProps {
  version: string;
  githubIssueURL: string;
  serviceVersions?: ServiceVersions;
  consent?: ConsentType;
  onClickRemoveConsent?(): void;
}

export type ServiceVersions = {
  nexus: string;
  admin: string;
  blazegraph: string;
  elasticsearch: string;
  iam: string;
  kg: string;
  storage: string;
};

const VersionInfo = (props: ServiceVersions) => {
  return (
    <>
      <h4 className="popover-title">Nexus Services</h4>
      {props.nexus && (
        <p>
          <label>Nexus</label> v{props.nexus}
        </p>
      )}
      <p>
        <label>Admin</label> v{props.admin}
      </p>
      <p>
        <label>IAm</label> v{props.iam}
      </p>
      <p>
        <label>Knowledge Graph</label> v{props.kg}
      </p>
      <h4 className="popover-title">Index Services</h4>
      <p>
        <label>Blaze Graph</label> v{props.blazegraph}
      </p>
      <p>
        <label>Elastic Search</label> v{props.elasticsearch}
      </p>
    </>
  );
};

const InformationContent = (props: InformationContentProps) => {
  return (
    <>
      <p>Nexus is Open Source and available under the Apache 2 License. </p>
      <p>v{props.version} </p>
      <p>
        Â© 2017-2020
        <a href="https://www.epfl.ch/" target="_blank">
          <img
            style={{ width: '3em', marginBottom: 3 }}
            src={epflLogo}
            alt="EPFL"
          />
        </a>
        {'| '}
        <a href="https://bluebrain.epfl.ch/" target="_blank">
          <span className="bbp-logo">Blue Brain Project</span>
        </a>
      </p>
      {props.serviceVersions ? (
        <VersionInfo {...props.serviceVersions} />
      ) : null}
      <p>
        <a href={documentationURL} target="_blank">
          <Icon type="book" /> Documentation
        </a>
        {' | '}
        <a href={props.githubIssueURL} target="_blank">
          <Icon type="github" /> Report Issue
        </a>
      </p>
      {
        <ConsentPreferences
          onClickRemove={props.onClickRemoveConsent}
          consent={props.consent}
        />
      }
    </>
  );
};
export interface HeaderProps {
  version: string;
  githubIssueURL: string;
  name?: string;
  token?: string;
  links?: React.ReactNode[];
  displayLogin?: boolean;
  children?: React.ReactChild;
  onLoginClick?(): void;
  visitHome?(): void;
  serviceVersions?: ServiceVersions;
  consent?: ConsentType;
  onClickRemoveConsent?(): void;
}

const Header: React.FunctionComponent<HeaderProps> = ({
  name,
  token,
  displayLogin = true,
  links = [],
  children,
  onLoginClick,
  version,
  githubIssueURL,
  visitHome,
  serviceVersions,
  consent,
  onClickRemoveConsent,
}) => {
  const studioView = useSelector((state: RootState) => state.config.studioView);

  const menu = (
    <Menu>
      {links.map((link, i) => (
        <Menu.Item key={i}>{link}</Menu.Item>
      ))}
    </Menu>
  );

  return (
    <header className="Header">
      <div className="selectors">{children}</div>
      <div className="logo-block">
        <a
          className="logo"
          href=""
          onClick={e => {
            if (visitHome) {
              e.preventDefault();
              visitHome();
            }
          }}
        >
          {
            // must add inline styling to prevent this big svg from
            // flashing the screen on dev mode before styles are loaded
          }
          <img style={{ height: '2em', width: '2em' }} src={logo} alt="Nexus" />
          <h1>Nexus</h1>
        </a>
      </div>
      <div className="menu-block">
        {studioView !== '' && [
          <a
            className="nav-item"
            href=""
            onClick={e => {
              if (visitHome) {
                e.preventDefault();
                visitHome();
              }
            }}
            key="admin-link"
          >
            Admin
          </a>,
          <a className="nav-item" href="nexus/studio" key="studio-link">
            Studio
          </a>,
        ]}
        {token && (
          <Copy
            textToCopy={token}
            render={(copySuccess, triggerCopy) => (
              <Button size="small" icon="copy" onClick={() => triggerCopy()}>
                {copySuccess ? 'Token copied!' : 'Copy token'}
              </Button>
            )}
          />
        )}

        {name ? (
          <Dropdown overlay={menu}>
            <a className="menu-dropdown ant-dropdown-link">
              {name} <Icon type="down" />
            </a>
          </Dropdown>
        ) : displayLogin ? (
          <a className="menu-dropdown ant-dropdown-link" onClick={onLoginClick}>
            login <Icon type="login" />
          </a>
        ) : null}
      </div>
    </header>
  );
};

export default Header;
