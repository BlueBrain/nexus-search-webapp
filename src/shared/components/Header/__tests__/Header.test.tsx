import * as React from 'react';
import { shallow, mount } from 'enzyme';
import Header, { HeaderProps } from '..';

const links: React.ReactNode[] = [
  <p>A p tag</p>,
  <a href="/somepage">A link to some page</a>,
];

const shallowHeader = shallow(<Header name="Mark Hamill" links={links} />);
const wrapper = mount(<Header name="Mark Hamill" links={links} />);

describe('Header component', () => {
  it('Should render correctly', () => {
    expect(shallowHeader).toMatchSnapshot();
  });

  it('Should have a logo block and a menu block', () => {
    expect(shallowHeader.find('.logo-block')).toBeTruthy();
    expect(shallowHeader.find('.menu-block')).toBeTruthy();
  });

  describe('Logo Block', () => {
    it('Should diplay the Nexus logo followed by the App name', () => {
      expect(wrapper.find('.logo')).toBeTruthy();
      expect(wrapper.find('h1').text()).toEqual('Nexus');
    });
  });

  describe('Menu Block', () => {
    it('Should display the user name followed by a space', () => {
      expect(wrapper.find('.menu-dropdown').text()).toEqual('Mark Hamill ');
    });

    it('Should display login link', () => {
      wrapper.setProps({ name: '' });
      expect(wrapper.find('.menu-dropdown').text()).toEqual('login ');
    });

    it('Should NOT display login link', () => {
      wrapper.setProps({ displayLogin: false });
      expect(wrapper.find('.menu-dropdown').children()).toHaveLength(0);
    });
  });
});
