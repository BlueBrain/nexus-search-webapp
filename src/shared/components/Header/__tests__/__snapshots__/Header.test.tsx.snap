// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should render correctly 1`] = `
<header
  className="Header"
>
  <div
    className="selectors"
  />
  <div
    className="logo-block"
  >
    <a
      className="logo"
      href=""
      onClick={[Function]}
    >
      <img
        alt="Nexus"
        src="/mock/path"
        style={
          Object {
            "height": "2em",
            "width": "2em",
          }
        }
      />
      <h1>
        Nexus
      </h1>
    </a>
  </div>
  <div
    className="menu-block"
  >
    <a
      className="nav-item"
      href=""
      key="admin-link"
      onClick={[Function]}
    >
      Admin
    </a>
    <a
      className="nav-item"
      href="nexus/studio"
      key="studio-link"
    >
      Studio
    </a>
    <Popover
      content={
        <InformationContent
          githubIssueURL=""
          serviceVersions={
            Object {
              "admin": "1.0",
              "blazegraph": "1.0",
              "elasticsearch": "1.0",
              "iam": "1.0",
              "kg": "1.0",
              "nexus": "1.0",
              "storage": "asda",
            }
          }
          version=""
        />
      }
      mouseEnterDelay={0.1}
      mouseLeaveDelay={0.1}
      overlayStyle={Object {}}
      placement="bottomRight"
      title="Information"
      transitionName="zoom-big"
      trigger="click"
    >
      <Button
        block={false}
        className="ui-header-info-button"
        ghost={false}
        htmlType="button"
        icon="info-circle"
        loading={false}
        size="small"
      />
    </Popover>
    <Dropdown
      mouseEnterDelay={0.15}
      mouseLeaveDelay={0.1}
      overlay={
        <Menu>
          <MenuItem>
            <p>
              A p tag
            </p>
          </MenuItem>
          <MenuItem>
            <a
              href="/somepage"
            >
              A link to some page
            </a>
          </MenuItem>
        </Menu>
      }
      placement="bottomLeft"
    >
      <a
        className="menu-dropdown ant-dropdown-link"
      >
        Mark Hamill
         
        <Icon
          type="down"
        />
      </a>
    </Dropdown>
  </div>
</header>
`;
